### PR TITLE
Feature/test tools

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+        integrationTestsEnvVars: [],
+        shouldPublishJars: true,
+        shouldPublishDockerImages: true,
+        dockerfileDir: 'docker',
+        buildContext: '.',
+        dockerImageRepositoryName: repositoryName,
+        preProductionVisibility: 'docker.io',
+        productionVisibility: 'docker.io')
+
+node('docker') { //any node would be fine
```
Comment (zguesmi) [inline Jenkinsfile:15]: Any node with minimum specs?

**Review by zguesmi — APPROVED** [review]
Thanks!

File: src/main/java/com/iexec/worker/tee/post/compute/App.java [hunk 2]
```diff
@@ -16,7 +16,10 @@
 @Slf4j
 public class App {
 
-    public static void main(String[] args) {
+    private App() {
+    }
+
+    public static void main() {
```
Comment (jbern0rd) [inline src/main/java/com/iexec/worker/tee/post/compute/App.java:22]: Why change java legal `main` method signature ?

File: Jenkinsfile (hunk 1 above)
Comment (jeremyjams) [reply Jenkinsfile:15]: Only required for processing sh command line yes

**Review by mcornaton — APPROVED** [review]
Thanks!

File: src/test/java/com/iexec/worker/tee/post/compute/AppTests.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+package com.iexec.worker.tee.post.compute;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class AppTests {
+
+    @Test
+    void main() {
+        Assertions.assertTrue(true);
```
Comment (jbern0rd) [inline src/test/java/com/iexec/worker/tee/post/compute/AppTests.java:26]: If we don't have test, I prefer we remove this

File: Jenkinsfile (hunk 1 above)
Comment (zguesmi) [reply Jenkinsfile:15]: I mean we need a minimum set of specs to run TEE builds.

Comment (jeremyjams) [reply Jenkinsfile:15]: Not here, `sconeBuildAllTee` already defines its own node specification (which we might upgrade btw)

File: build.gradle [hunk 4]
```diff
@@ -19,10 +9,10 @@ buildscript {
...
 
 plugins {
-    // Apply the java plugin to add support for Java
     id 'java'
-    // Apply the application plugin to add support for building a CLI application
     id 'application'
+    id 'jacoco'
+    id 'org.sonarqube' version '3.3'
     id 'com.palantir.docker' version '0.26.0'
     id 'com.github.johnrengelman.shadow' version '6.1.0'
```
Comment (jbern0rd) [inline build.gradle:17]: Are `com.github.jengelman.gradle.plugins:shadow:4.0.4` and `id 'com.github.johnrengelman.shadow' version '6.1.0'` 2 versions of the same plugin ?
If true, the entire `buildscript` block should be removable.

File: Jenkinsfile (hunk 1 above)
Comment (zguesmi) [reply Jenkinsfile:15]: Ok thanks for the clarification.

File: src/main/java/com/iexec/worker/tee/post/compute/App.java (hunk 2 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/worker/tee/post/compute/App.java]: Auto-correction, fixed, thanks for heads-up!

File: build.gradle (hunk 4 above)
Comment (jeremyjams) [reply build.gradle:7]: True, I updated bunch of lines for that, thanks :)

**Review by jbern0rd — APPROVED** [review]

**Pull request merged**